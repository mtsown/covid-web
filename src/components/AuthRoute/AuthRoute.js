import React from 'react';
import { Route, Redirect } from 'react-router-dom';

function AuthRoute({ component: Component, ...rest }) {
  const token = localStorage.getItem('userToken');

  return (
    <Route
      {...rest}
      render={() => (token === null ? <Component /> : <Redirect to="/" />)}
    />
  );
}

export default AuthRoute;
