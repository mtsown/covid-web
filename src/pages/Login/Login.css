.login {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100vw;
  height: 100vh;
  background-image: url("../../assets/images/common-background.jpg");
  background-repeat: repeat;
  background-size: 100% 100%;
}

.login.disabled-login {
  pointer-events: none;
}

.login h1.ant-typography {
  color: var(--white);
  margin: 1rem 0 1rem 0;
}

.login .login-form {
  background-color: var(--white);
  min-width: 300px;
  padding: 2rem;
  border: none;
  border-radius: 1rem;
  box-shadow: 5px 10px 20px var(--shadow-color);
}

.login .ant-form .ant-row .ant-form-item-control-input-content {
  display: flex;
  justify-content: center;
}

.login .ant-form .site-form-item-icon {
  color: var(--shadow-color);
}

.login .ant-typography.ant-typography-danger {
  margin-bottom: 1rem;
}
