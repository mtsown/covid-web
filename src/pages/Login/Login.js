import React, { useState, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link, useHistory } from 'react-router-dom';

import { Row, Form, Input, Button, Typography, message } from 'antd';
import { UserOutlined, LockOutlined } from '@ant-design/icons';
import { RiVirusLine } from 'react-icons/ri';
import { toggleGlobalLoading } from '../../redux/appSlice';
import CryptoJS from 'crypto-js';
import './Login.css';

function Login() {
  const [, forceUpdate] = useState();
  const [warningMessage, setWarningMessage] = useState('');

  const isLoading = useSelector(state => state.app.isLoading);
  const usersList = useSelector(state => state.users);

  const dispatch = useDispatch();
  const history = useHistory();
  const { Text, Title } = Typography;
  const [form] = Form.useForm();
  const key = 'updatable';

  useEffect(() => {
    forceUpdate({});
  }, []);

  const handleClearWarning = () => {
    setWarningMessage('');
  };

  const openMessage = () => {
    message.loading({ content: 'Logging in...', key });
    dispatch(toggleGlobalLoading());
    setTimeout(() => {
      message.success({ content: 'Welcome back!', key, duration: 2 });
      dispatch(toggleGlobalLoading());
      history.push('/');
    }, 1500);
  };

  const handleSubmitLogin = (credentials) => {
    const filteredUsersById = usersList.filter(user => user.id === credentials.id);
    if (filteredUsersById.length > 0) {
      if (filteredUsersById[0].password === credentials.password) {
        const token = CryptoJS.AES.encrypt(JSON.stringify(filteredUsersById[0]), 'SonNT72').toString();
        localStorage.setItem('userToken', token);
        openMessage();
      } else setWarningMessage('Password incorrect!');
    } else setWarningMessage('Username does not exist!');
  };

  return (
    <div className={isLoading ? 'login disabled-login' : 'login'}>
      <div className="virus-logo">
        <RiVirusLine />
      </div>
      <Title>Sign in</Title>
      <div className="login-form">
        <Form form={form} name="horizontal_login" layout="vertical" onFinish={handleSubmitLogin} size="large">
          <Form.Item
            name="id"
            rules={[
              {
                required: true,
                message: 'Please input your username!'
              }
            ]}
          >
            <Input
              prefix={<UserOutlined className="site-form-item-icon" />}
              placeholder="Username"
              onChange={handleClearWarning} />
          </Form.Item>
          <Form.Item
            name="password"
            rules={[
              {
                required: true,
                message: 'Please input your password!'
              }
            ]}
          >
            <Input.Password
              prefix={<LockOutlined className="site-form-item-icon" />}
              type="password"
              placeholder="Password"
              onChange={handleClearWarning}
            />
          </Form.Item>
          {warningMessage !== '' ? <Row justify="center"><Text type="danger">{warningMessage}</Text></Row> : null}
          <Form.Item shouldUpdate={true}>
            {() => (
              <Button
                type="primary"
                htmlType="submit"
                disabled={
                  !form.isFieldsTouched(true) ||
                  form.getFieldsError().filter(({ errors }) => errors.length).length
                }
              >
                Log in
              </Button>
            )}
          </Form.Item>
        </Form>
        <Row justify="center"><Text>Don't have an account? <Link to="/register">Sign up</Link></Text></Row>
      </div>
    </div>
  );
}

export default Login;
