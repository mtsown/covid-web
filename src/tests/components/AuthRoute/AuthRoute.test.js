import React from 'react';
import { shallow, mount } from 'enzyme';
import AuthRoute from 'components/AuthRoute/AuthRoute.js';
import Login from 'pages/Login/Login.js';

describe('AuthRoute', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<AuthRoute />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly with props', () => {
    const wrapper = shallow(<AuthRoute path="/login" component={Login} />);
    expect(wrapper).toMatchSnapshot();
  });
});
