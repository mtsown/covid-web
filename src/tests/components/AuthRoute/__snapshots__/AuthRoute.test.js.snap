// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthRoute should render correctly 1`] = `
<Route
  render={[Function]}
/>
`;

exports[`AuthRoute should render correctly with props 1`] = `
<Route
  path="/login"
  render={[Function]}
/>
`;
